jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,430 @@
+        message = "$Servo,{angle0},{angle1},{angle2}"
+        message = message.format(angle0=struct.angle0, angle1=struct.angle1, angle2=struct.angle2)
+        print(message)
+        while(len(message) < UART_TRANSMIT_MSG_LEN):
+            message += ","
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+
+    def carousel_openloop_transmit(self, channel, msg):
+        struct = CarouselOpenLoopCmd.decode(msg)
+
+        if (self.last_openloop_cmd == struct.throttle):
+            return
+        print("Received Carousel (OL) Cmd")
+        self.last_openloop_cmd = struct.throttle
+        # parse data into expected format
+        message = "$OpenCarousel,{throttle}"
+        message = message.format(throttle=struct.throttle)
+        print(message)
+        while(len(message) < UART_TRANSMIT_MSG_LEN):
+            message += ","
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+
+    def carousel_closedloop_transmit(self, channel, msg):
+        self.last_openloop_cmd = -1
+        struct = CarouselPosition.decode(msg)
+        print("Received Carousel (CL) Cmd")
+        # parse data into expected format
+        message = "$Carousel,{position}"
+        message = message.format(position=struct.position)
+        print(message)
+        while(len(message) < UART_TRANSMIT_MSG_LEN):
+            message += ","
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+
+    async def receive(self, lcm):
+        spectral = SpectralData()
+        thermistor = ThermistorData()
+        triad = SpectralData()
+        carousel = CarouselPosition()
+        heater = Heater()
+        heater_auto_shutdown = HeaterAutoShutdown()
+
+        # Mark TXT as always seen because they are not necessary
+        seen_tags = {tag: False if not tag == 'TXT' else True
+                     for tag in self.NMEA_HANDLE_MAPPER.keys()}
+        while True:
+            # Wait for all tags to be seen
+            while (not all(seen_tags.values())):  # TODO -
+                try:
+                    error_counter = 0  # TODO - DELETE
+                    tx = self.ser.readline()
+                    msg = str(tx)
+                except Exception as e:
+                    print("Errored")
+                    if error_counter < self.max_error_count:
+                        error_counter += 1
+                        print(e)
+                        await asyncio.sleep(self.sleep)
+                        continue
+                    else:
+                        raise e
+                match_found = False
+                for tag, func in self.NMEA_HANDLE_MAPPER.items():
+                    if tag in msg:  # TODO - why do we have tag in msg, func is not even used
```

Comment:
no....use func!!!! instead of self.spectral_handler etc
if tag == < string > 
func(msg.decode(), < struct >)